pkg/openai/count.go

```diff
@@ -54,22 +67,44 @@ func dropMessagesOverCount(maxTokens int, msgs []openai.ChatCompletionMessage) (
 	if withinBudget == len(msgs)-1 {
 		// We are going to drop all non system messages, which seems useless, so just return them
 		// all and let it fail
-		return msgs
+		return msgs, nil
 	}
 
-	return append(result, msgs[withinBudget:]...)
+	return append(result, msgs[withinBudget:]...), nil
 }
 
-func countMessage(msg openai.ChatCompletionMessage) (count int) {
-	count += len(msg.Role)
-	count += len(msg.Content)
+func countMessage(msg openai.ChatCompletionMessage) (int, error) {
+	tiktoken.SetBpeLoader(tiktoken_loader.NewOfflineLoader())
+	encoding, err := tiktoken.GetEncoding("o200k_base")
```

Comment:
can we set the loader once on package init?

Reply:
Yep, good call. Just pushed the change for this